charts/monitoring/templates/clusterrole.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.adminDeploy }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: prometheus-cluster-role-testing
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+      - services
+      - endpoints
+    verbs:
+      - get
+      - watch
+      - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-cluster-role-binding-testing
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus-cluster-role-testing
+subjects:
+  - kind: ServiceAccount
+    name: default
```

Comment:
The name of the service account should be conditional. .Values.prometheus.serviceAccounts.server.name